packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts

```diff
@@ -335,48 +336,22 @@ function useTreeInternal({
   };
 }
 
-function generateTreeStructure(parentNodeId: string | undefined, model: TreeModel): Array<PresentationTreeNode> | undefined {
+function generateTreeStructure(parentNodeId: string | undefined, model: TreeModel): Array<PresentationHierarchyNode> | undefined {
   const currentChildren = model.parentChildMap.get(parentNodeId);
   if (!currentChildren) {
     return undefined;
   }
 
   return currentChildren
     .map((childId) => model.idToNode.get(childId))
-    .filter((node): node is TreeModelNode => !!node)
-    .map<PresentationTreeNode>((node) => {
-      if (isTreeModelHierarchyNode(node)) {
-        return createPresentationHierarchyNode(node, model);
-      }
-
-      if (node.type === "ResultSetTooLarge") {
-        return {
-          id: node.id,
-          parentNodeId,
-          type: node.type,
-          resultSetSizeLimit: node.resultSetSizeLimit,
-        };
-      }
-
-      if (node.type === "NoFilterMatches") {
-        return {
-          id: node.id,
-          parentNodeId,
-          type: node.type,
-        };
-      }
-
-      return {
-        id: node.id,
-        parentNodeId,
-        type: node.type,
-        message: node.message,
-      };
+    .filter((node): node is TreeModelHierarchyNode => !!node)
+    .map<PresentationHierarchyNode>((node) => {
+      return createPresentationHierarchyNode(node, model);
     });
 }
 
 function createPresentationHierarchyNode(modelNode: TreeModelHierarchyNode, model: TreeModel): PresentationHierarchyNode {
-  let children: Array<PresentationTreeNode> | undefined;
+  let children: Array<PresentationHierarchyNode> | undefined;
```

Comment:
this is probably going to be a stupid question... but do we need to have `TreeModelHierarchyNode` and `PresentationHierarchyNode`? the two look very similar...

Reply:
When we load nodes we still load modelNodes and errorNodes seperatly. `TheeModelHierarchyNode` also contain some loading information such as `hierarchyLimit` and `instanceFilter`, which would make `PresentationHierarchyNode` overloaded with information it does not need. Having them merged into one I think would add unnecessary complexity.